docs/api/utility.rst

```diff
@@ -0,0 +1,325 @@
+.. # Copyright (c) 2017-2019, Lawrence Livermore National Security, LLC and
+.. # other BLT Project Developers. See the top-level COPYRIGHT file for details
+.. # 
+.. # SPDX-License-Identifier: (BSD-3-Clause)
+
+Utility Macros
+==============
+
+
+General Utility Macros
+----------------------
+
+
+blt_list_append
+~~~~~~~~~~~~~~~
+
+.. code-block:: cmake
```

Comment:
Would it be clearer if the prototypes/signatures had a caption, e.g. ``:caption: **Signature** `` ?

Reply:
Due to sphinx formatting it was weird to have it there...